exposing CPU/GPU Copy ops

wat3rBro opened the pull request:
Summary: expose CPU/GPU copy ops

Test Plan: TBA

Differential Revision: D19405856

smessmer commented on caffe2/core/export_caffe2_op_to_c10.h:
```diff
@@ -165,6 +165,18 @@ inline FunctionSchema make_function_schema_for_c10(const char* schema_str) {
+  }                                                                          \
+  }
+
 #define C10_EXPORT_CAFFE2_OP_TO_C10_CPU(                                     \
```
Can you rewrite this one as calls to`_SCHEMA_ONLY` and `_KERNEL_ONLY` to avoid the duplication?

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit f5428d73:


* **1/2** failures introduced in this PR
* **1/2** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=f5428d73768d5a478fa6941961cbabcd6eea85f6).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4261107) pytorch_linux_xenial_cuda10_1_cudnn7_py3_gcc7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86683784) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86683784)) :snowflake:
<details>
<summary>
<code>Jan 15 22:56:02   Error generating file</code>
</summary>

```
Jan 15 22:56:01  
Jan 15 22:56:01 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Jan 15 22:56:01  
Jan 15 22:56:01 [ 78%] Building NVCC (Device) object caffe2/CMakeFiles/torch_cuda.dir/operators/torch_cuda_generated_elementwise_linear_op.cu.o 
Jan 15 22:56:02 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Jan 15 22:56:02  
Jan 15 22:56:02 /var/lib/jenkins/workspace/caffe2/operators/copy_op.cu(58): error: name followed by "::" must be a class or namespace name 
Jan 15 22:56:02  
Jan 15 22:56:02 1 error detected in the compilation of "/tmp/tmpxft_00000ae8_00000000-6_copy_op.cpp1.ii". 
Jan 15 22:56:02 CMake Error at torch_cuda_generated_copy_op.cu.o.Release.cmake:281 (message): 
Jan 15 22:56:02   Error generating file 
Jan 15 22:56:02   /var/lib/jenkins/workspace/build/caffe2/CMakeFiles/torch_cuda.dir/operators/./torch_cuda_generated_copy_op.cu.o 
Jan 15 22:56:02  
Jan 15 22:56:02  
Jan 15 22:56:02 make[2]: *** [caffe2/CMakeFiles/torch_cuda.dir/operators/torch_cuda_generated_copy_op.cu.o] Error 1 
Jan 15 22:56:02 make[2]: *** Waiting for unfinished jobs.... 
Jan 15 22:56:02 caffe2/CMakeFiles/torch_cuda.dir/build.make:1309: recipe for target 'caffe2/CMakeFiles/torch_cuda.dir/operators/torch_cuda_generated_copy_op.cu.o' failed 
Jan 15 22:56:03 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Jan 15 22:56:03  
Jan 15 22:56:03 /var/lib/jenkins/workspace/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity 
Jan 15 22:56:03  
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

wat3rBro committed:
exposing CPU/GPU Copy ops (#32248)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32248

expose CPU/GPU copy ops

Test Plan: buck test mode/dev-nosan caffe2/caffe2/python/operator_test:torch_integration_test

Differential Revision: D19405856

fbshipit-source-id: 8ecd3cb144e96d4b72bb0a623ce02b2c9e51ff49

houseroad reviewed (approved):
shipit
